src/fiber/network.rs

```diff
@@ -2758,12 +2759,59 @@ where
         channel_id: Hash256,
         command: ChannelCommand,
     ) -> crate::Result<()> {
-        match self.channels.get(&channel_id) {
-            Some(actor) => {
-                actor.send_message(ChannelActorMessage::Command(command))?;
-                Ok(())
+        match command {
+            // Need to handle the force shutdown command specially because the ChannelActor may not exist when remote peer is disconnected.
+            ChannelCommand::Shutdown(shutdown, rpc_reply) if shutdown.force => {
+                match self.store.get_channel_actor_state(&channel_id) {
+                    Some(mut state) => {
+                        match state.state {
+                            ChannelState::ChannelReady() => {
+                                debug!("Handling force shutdown command in ChannelReady state");
+                            }
+                            ChannelState::ShuttingDown(flags) => {
+                                debug!("Handling force shutdown command in ShuttingDown state, flags: {:?}", &flags);
+                            }
+                            _ => {
+                                return Err(Error::ChannelError(
+                                    ProcessingChannelError::InvalidState(format!(
+                                        "Handling force shutdown command invalid state {:?}",
+                                        &state.state
+                                    )),
+                                ));
+                            }
+                        };
+
+                        state.check_shutdown_fee_rate(shutdown.fee_rate, &shutdown.close_script)?;
+
+                        // when channel is in ChannelReady or ShuttingDown state, the latest_commitment_transaction should exist
+                        let transaction = state.latest_commitment_transaction.clone().unwrap();
+                        self.network
+                            .send_message(NetworkActorMessage::new_event(
+                                NetworkActorEvent::CommitmentTransactionPending(
+                                    transaction,
+                                    channel_id,
+                                ),
+                            ))
+                            .expect(ASSUME_NETWORK_ACTOR_ALIVE);
+
+                        state.update_state(ChannelState::ShuttingDown(
+                            ShuttingDownFlags::WAITING_COMMITMENT_CONFIRMATION,
+                        ));
+                        self.store.insert_channel_actor_state(state);
+
+                        let _ = rpc_reply.send(Ok(()));
```

Comment:
We didn't return error to the client if above code runs into an error condition.